pkg/json.go

```diff
@@ -122,7 +123,8 @@ func (r *ScorecardResult) AsJSON(showDetails bool, logLevel log.Level, writer io
 
 // AsJSON2 exports results as JSON for new detail format.
 func (r *ScorecardResult) AsJSON2(showDetails bool,
-	logLevel log.Level, checkDocs docs.Doc, writer io.Writer,
+	showMaintainersAnnotation bool, logLevel log.Level,
+	checkDocs docs.Doc, writer io.Writer,
```

Comment:
minor:
use a configuration struct maybe? here's a similar one for probe format for example

https://github.com/ossf/scorecard/blob/5a96bddb3aaffeb819d8653051b259dbfd08ffa2/pkg/probe.go#L34-L42

Reply:
My feeling during the code change is that it didn't get better. It's too verbose and created new types only to create the options structure. I'd rather keep the long function parameters for now and refactor the whole function parameters calls some other time. But if you have a strong preference for this format, I can make the changes.

Reply:
part of it is a desire to be able to extend the function in the future in a non-breaking way. Modifying function signatures is a breaking change. So long term, breaking it once means not breaking it again.

it doesn't need to be included in this PR, but is something I see value in

Reply:
Got it. One question though, what you have in mind is something like:
```
struct AsJSON2ResultOption {
  ShowDetails bool
  ShowMaintainersAnnotation bool
}

func (r *ScorecardResult) AsJSON2(o *AsJSON2ResultOption,
     logLevel log.Level, checkDocs docs.Doc, writer io.Writer)
```
with `AsJSON2ResultOption` encapsulating only the result options or encapsulating other fields too like `logLevel` and `checkDocs`?

Reply:
`checkDocs` is fundamental to the call. so I would lean towards something like:
```go
struct AsJSON2Option {
  Details bool
  MaintainersAnnotation bool
  LogLevel log.Level
}

func (r *ScorecardResult) AsJSON2(writer io.Writer, checkDocs docs.Doc, o *AsJSON2Option)
 ```

Reply:
Ack, thanks for the response!

Reply:
Done for AsJSON2 and AsString.